compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/methodgen/MethodGen.java

```diff
@@ -905,6 +911,7 @@ private void generateFrameClassFieldUpdateByTypeTag(MethodVisitor mv, String fra
                 break;
             case TypeTags.TABLE:
                 mv.visitVarInsn(ALOAD, index);
+                mv.visitTypeInsn(CHECKCAST, TABLE_VALUE_IMPL);
```

Comment:
Why do we need the checkcast only for the table case? Shall we check whether there is more correct alternative way to fix this issue? What is the difference between record/map/tuple/array case and the table case?

Reply:
I think we might need to handle for other types as well

Reply:
I checked the other value cases. We use API interface variables to store such values (`ArrayValue`, `MapValue` etc.). And these didn't break. 
But for tables, we use `TableValueImpl` values. And they are breaking.